rsconnect/subprocesses/inspect_environment.py

```diff
@@ -60,21 +61,21 @@ class EnvironmentException(Exception):
     pass
 
 
-def detect_environment(dirname: str, force_generate: bool = False) -> EnvironmentData:
+def detect_environment(dirname: str, requirements_file: Optional[str] = "requirements.txt") -> EnvironmentData:
     """Determine the python dependencies in the environment.
 
     `pip freeze` will be used to introspect the environment.
 
     :param: dirname Directory name
-    :param: force_generate Force the generation of an environment
+    :param: requirements_file The requirements file to read. If None, generate using pip freeze.
```

Comment:
Thoughts on if we could incorporate the spirit of https://github.com/posit-dev/rsconnect-python/pull/680 into this change too? `pip freeze` has never been satisfying to most of our users.

Reply:
> happy to pick up that work in the context of that PR if you can't move it forward

Thank you, I would very much appreciate that!